pytorch_lightning/callbacks/progress.py

```diff
@@ -496,8 +499,8 @@ def _update_bar(self, bar: Optional[tqdm]) -> None:
 
 
 def convert_inf(x: Optional[Union[int, float]]) -> Optional[Union[int, float]]:
-    """ The tqdm doesn't support inf values. We have to convert it to None. """
-    if x == float('inf'):
+    """ The tqdm doesn't support inf/nan values. We have to convert it to None. """
+    if x is None or math.isinf(x) or math.isnan(x):
```

Comment:
it's probably nan here because some arithmetic with inf produces it. 
did you observe where that happens?

Reply:
my guess is `self.total_train_batches + self.total_val_batches` in `on_train_batch_end` since we were missing the float(inf) check there? i'll see if i can pinpoint this better

Reply:
no it can't be because these numbers are all positive. it must be one of these cases where the arithmetic is undefined:
inf - inf
inf / inf
x / 0

Reply:
@awaelchli 

following the repro from #7303 

```
    ckpt_callback = ModelCheckpoint(monitor="val_log", save_top_k=1, mode="max", verbose=False)
    epoch_cb = Counter()
    epochs = 2
    trainer = Trainer(
        default_root_dir=tmpdir,
        num_sanity_val_steps=0,
        max_epochs=epochs,
        callbacks=[epoch_cb, ckpt_callback],
        limit_train_batches=3,
    )
    model = DummyModel()
    batch_size = 8
    train_dl = DataLoader(dataset=RandomIterableDataset(32, 128), batch_size=batch_size)
    val_dl = DataLoader(dataset=RandomIterableDataset(32, 128), batch_size=batch_size)
    trainer.fit(model, train_dataloader=train_dl, val_dataloaders=val_dl)
```

in the progress callback:
- total_train_batches=3
- total_val_batches=inf
- we only check if total_train_batches != inf, not val_batches
- trainer.val_check_batch is inf
- val_checks_per_epoch = 3 / inf -> 0.0
- total_val_batches = 0.0 * inf -> nan

Reply:
true, 0*inf is another case :))

do you think we should handle the logic here rather than in the progress bar code?

Reply:
We can do both, just in case non-number inputs are somehow passed to tqdm in another path. I also updated the check inside the progress callback in case total_val_batches is also float inf